rust/src/tx_builder.rs

```diff
@@ -240,6 +256,23 @@ impl TransactionBuilder {
                         }
                     }
                 }
+                // Phase 3: add extra inputs needed for fees
+                // We do this at the end because this new inputs won't be associated with
+                // a specific output, so the improvement algorithm we do above does not apply here.
+                if input_total.partial_cmp(&output_total).unwrap().eq(&Ordering::Less) {
```

Comment:
1. Why using `partial_cmp` here?
2. Unchecked unwrap doesn't look good.

If these two types are both `Value` I think you can even just use `<`, e.g. see how `while added < needed` is used above

Reply:
Ah, that was a bad copy paste from my end 😅
Just pushed a version with a regular `>` comparison